src/libcore/iter.rs

```diff
@@ -369,6 +369,25 @@ pub trait Iterator {
     /// `None` here means that either there is no known upper bound, or the
     /// upper bound is larger than `usize`.
     ///
+    /// # Safety
```

Comment:
Is this safety? I think it's just about the protocol expected from the method. Maybe implementation notes. @steveklabnik, do we have a precedent?